command/agent/dns.go

```diff
@@ -337,6 +337,14 @@ PARSE:
 		node := strings.Join(labels[:n-1], ".")
 		d.nodeLookup(network, datacenter, node, req, resp)
 
+	case "query":
+		if len(labels) == 1 {
```

Comment:
Not sure why this case and the one above it don't just use `n` instead of doing `len()` again